lib/Conversion/TorchToTosa/TorchToTosa.cpp

```diff
@@ -4218,6 +4218,80 @@ class ConvertAtenCloneOp : public OpConversionPattern<AtenOpT> {
   }
 };
 
+template <>
+LogicalResult ConvertAtenOp<AtenConstantPadNdOp>::matchAndRewrite(
+    AtenConstantPadNdOp op, OpAdaptor adaptor,
+    ConversionPatternRewriter &rewriter) const {
+  Location loc = op.getLoc();
+  Value self = adaptor.getSelf();
+  auto selfTy = self.getType().cast<RankedTensorType>();
+  auto selfElemTy = selfTy.getElementType();
+  auto selfShape = selfTy.getShape();
+  //auto selfTensorTy = RankedTensorType::get(makeShapeLLVMCompatible(selfShape), selfElemTy);
+  int64_t rank = selfTy.getRank();
+
+  // Pattern match against the op's original operands, because otherwise we
+  // will get the lowered version of the operands which is harder to pattern
+  // match.
+  SmallVector<int64_t> padInts;
+  if (!matchPattern(op.getPad(), m_TorchListOfConstantInts(padInts)))
+    return rewriter.notifyMatchFailure(
+        op, "only support constant int pad ranges");
+  uint64_t padRank = padInts.size() / 2;
+  if (padRank * 2 != padInts.size())
+    return rewriter.notifyMatchFailure(op, "pad range size is not even");
+  if (rank < 0 || padRank > (uint64_t)rank)
+    return rewriter.notifyMatchFailure(op, "padding exceeds tensor rank");
+
+  // Initialize low/high paddings with the dims that should not be padded.
+  SmallVector<int64_t, 4> lowPadding(/*Size=*/rank - padRank, /*Value=*/0);
+  SmallVector<int64_t, 4> highPadding(/*Size=*/rank - padRank, /*Value=*/0);
+  // Add the requested padding - note op.pad() is highest dim first ordered
+  // pairs of low,high.
+  for (uint64_t i = padRank; i > 0; --i) {
+    lowPadding.push_back(padInts[i * 2 - 2]);
+    highPadding.push_back(padInts[i * 2 - 1]);
+  }
+
+  llvm::SmallVector<int64_t, 8> translatePadsList;
+
+  const unsigned int dimSize = lowPadding.size();
+  for (unsigned int i = 0; i < dimSize; i++) {
+    translatePadsList.push_back(lowPadding[i]);
+    translatePadsList.push_back(highPadding[i]);
+  }
+
+  DenseElementsAttr paddingAttr = DenseIntElementsAttr::get(
+      RankedTensorType::get({dimSize, 2}, rewriter.getI64Type()),
+      translatePadsList);
+
+  Value padsList1 = rewriter.create<mlir::tosa::ConstOp>(
+      loc, paddingAttr.getType(), paddingAttr);
+
+  Value padValue = adaptor.getValue();
+  Operation *padOp = padValue.getDefiningOp();
+  if (padOp != nullptr) {
```

Comment:
When could this case happen?

Reply:
The only case I think it can happen is that the constant value is directly used in the torch.aten.constant_pad_nd operation. Therefore, there is no definingOp for the constant.

Reply:
I'm not sure whether that case would be valid IR. Maybe you could show that IR case? (E.g. in https://xiractwebsrv01.xilinx.com/slxexplorer/z/bK1eYE and generate a new "short link")

Reply:
Thanks Matthias.  torch.aten.constant_pad_nd requires a SSA operand for the padding value. So, the case I mentioned wouldn't be a valid IR. Link https://xiractwebsrv01.xilinx.com/slxexplorer/ is very helpful for checking that matter. Thanks for sharing it. I've removed the redundant check now.

Reply:
Ok, great that you find it helpful.

For the remaining code
```
 Value padValue = adaptor.getValue();
  Operation *padOp = padValue.getDefiningOp();
  padValue = padOp->getOperand(0);

  Type padValueType = padValue.getType();

  Value padTensor;
  if (failed(torchScalarToTosaTensor(rewriter, op.getOperation(), padValue,
                                  padTensor, selfElemTy, {})))
```
I was under the impression that
```
  Value padTensor;
  if (failed(torchScalarToTosaTensor(rewriter, op.getOperation(), adaptor.getValue(),
                                  padTensor, selfElemTy, {})))
```
would be enough. Could you check that, please?

Reply:
we need following statements to get the padding value.

   Operation *padOp = padValue.getDefiningOp();
  padValue = padOp->getOperand(0);
  
otherwise, for following value
%float-Inf = torch.constant.float 0xFFF0000000000000
torchScalarToTosaTensor cannot map %float-inf to float32 for lowering it to tosa.pad.